src/http-api.ts

```diff
@@ -274,7 +274,14 @@ export class HttpApi<S extends Schema> extends EventEmitter {
             errorCode: `ERROR_HTTP_${response.statusCode}`,
             message: response.body,
           };
-    return new HttpApiError(error.message, error.errorCode);
+
+    return error.message.includes('<html ')
+      ? new HttpApiError(
```

Comment:
AFAIK there's no endpoint that's expected to return HTML but since we have access to the res headers here, can you make it check that `content-type === text/html` is true instead if searching for html tags?

![Screenshot 2023-11-13 at 18 52 54](https://github.com/jsforce/jsforce/assets/6853656/94569efe-44b6-420a-83d1-e4a6a1404b87)

Also, do you think it's worth keeping the HTML body in the error? we could just remove it and `logger.debug` it here, if someone is ever interested they can just set `JSFORCE_LOG_LEVEL=debug`

Reply:
In the case of the 420 org error page, it's a lot of salesforce-branded SVG plotting and js scripts.

I was thinking there might be some simple error pages that people were parsing today, such that logger wouldn't be enough.  Logging might be a good idea in addition to keeping the error?

I don't know.  Hard call.

Reply:
pushing up the header-based check with logger